test/eth.subscribe.ganache.js

```diff
@@ -35,6 +35,61 @@ describe('subscription connect/reconnect', function() {
         });
     });
 
+    it('subscribes with a callback', function(){
+        let subscription;
+
+        return new Promise(async function (resolve) {
+            subscription = web3.eth
+                .subscribe('newBlockHeaders', function(err, result){
+                    assert(result.parentHash);
+                    subscription.unsubscribe(); // Stop listening..
+                    resolve();
+            });
+         })
+    });
+
+    it('resubscribes', function(){
+        let subscription;
+
+        return new Promise(async function (resolve) {
+            subscription = web3.eth.subscribe('newBlockHeaders');
+            subscription.unsubscribe();
+            subscription.resubscribe()
+
+            subscription.on('data', function(result){
```

Comment:
Shouldn't we register here the listener directly after the call of ``web3.eth.subscribe`` to be sure the ``data`` event listener is kept?

Reply:
Rewrote this in 21d854c362475a694d79f32dc38dbfa4272e41f9. Is that closer to what you think it should be?

Reply:
Not exactly what I thought about. We should be able to register one ``data`` listener and this listener should still get triggered after the call of ``resubscribe``. It is probably easier to test this with the ``connected`` event. I've also noticed that we only have to call ``unsubscribe`` once because the ``resubscribe`` is already unsubscribing the subscription.

Pseudo-code: 
``` javascript
it('...', (done) => {
  stage = 0;
  subscription = web3.eth.subscribe('newBlockHeaders');
  subscription.on('data',() => {
    if (stage === 0) {
      subscription.resubscribe();
      stage = 1;
      return;
    }

    assert(result.parentHash);
    subscription.unsubscribe(); // Stop listening..
    done();
  });
});

```

Reply:
@nivida Have added as a separate case in ac880f656b564bf275ff2d417c3f2d62830a9ea2. 

In your view, are these actually different scenarios?

If not, what is the intended use-case of the `resubscribe` method?

Reply:
> In your view, are these actually different scenarios?

Yes, because the ``resubscribe`` is doing an ``unsubscribe`` as well and doesn't remove the event listeners of the current subscription. Because ``unsubscribe`` and ``resubscribe`` is called in your case is it doing a double ``unsubscribe`` call once with removing of the listeners and once without.